test3.js

```diff
@@ -0,0 +1,191 @@
+  try {
+    const { action, listObj, deliveryQ, setOrgPaginateAndOnboard, schedulerEnabled, updateHookDelivery, getCommitsFromDB } = params;
+    const {
+      delivery: { org, orgId }
+    } = deliveryQ;
+
+    const RepoModel = await GetRepoModel(org, schedulerEnabled);
+
+    const { repo_name: name, repo_id: id } = listObj;
+    Logger.log('info', `add or update repository from hooks : ${name}`);
+    const query = { repo_id: id };
+
+    const updateParams = { RepoModel, deliveryQ, updateHookDelivery };
+    if (action === 'deleted') {
+      const res = await DeleteRepository({ ...updateParams, query });
+      return res;
+    }
+
+    if (action !== 'created') {
+      // get commits data for the repo and update created at based on commits
+      const [commit] = await getCommitsFromDB({ orgName: org, repoId: id, repoName: name });
+      if (commit && new Date(commit.date) < new Date(listObj.created_at)) listObj.created_at = commit.date;
+    }
+
+    const res = await UpdateRepository({ ...updateParams, query, update: listObj, options: { upsert: true, setDefaultsOnInsert: true } });
+    if (res && action === 'created') await setOrgPaginateAndOnboard({ orgId });
+    return res;
+  } catch (exc) {
+    Logger.log('error', `Error in AddOrModifiyRepoInDB in ${path.basename(__filename)}: ${JSON.stringify(exc)}`);
+    throw exc;
+  }
+};
+
+/**
+ * Update repository updated_at from hooks
+ *
+ * @async
+ * @function SetRepoUpdateAtInDB
+ * @params {object}
+ * @returns {object} - array of objects
+ * @author dev-team
+ */
+
+const SetRepoUpdateAtInDB = async params => {
+  try {
+    Logger.log('info', `Update repository updated_at from hooks`);
+    const { id, updateAt, org } = params;
+
+    const RepoModel = await GetRepoModel(org);
+
+    const query = { repo_id: id };
+
+    const repo = await RepoModel.findOne(query);
+    if (!repo) {
+      Logger.log('info', `Update repository updated_at from hook rebo ${id} not found`);
+      return;
+    }
+
+    const update = { $set: { updated_at: updateAt } };
+    await RepoModel.updateOne(query, update, { upsert: false });
+    return;
+  } catch (exc) {
+    Logger.log('error', `Error in SetRepoUpdateAtInDB in ${path.basename(__filename)}: ${JSON.stringify(exc)}`);
+    throw exc;
+  }
+};
+
+/**
+ * Check repo deleted or not in hooks
+ *
+ * @async
+ * @function GetIsRepoDeletedInDB
+ * @params {object}
+ * @returns {object}
+ * @author dev-team
+ */
+
+const GetIsRepoDeletedInDB = async params => {
+  try {
+    Logger.log('info', `Check repo deleted or not in hooks`);
+    const { orgName, repoId } = params;
+
+    const RepoModel = await GetRepoModel(orgName);
+
+    const query = { repo_id: repoId };
+
+    const res = await RepoModel.findOne(query, { _id: 0, deleted: 1 });
+    return res ? res.deleted : res;
+  } catch (exc) {
+    Logger.log('error', `Error in GetIsRepoDeletedInDB in ${path.basename(__filename)}: ${JSON.stringify(exc)}`);
```

Comment:
Codacy has a fix for the issue: [Replace `'error',·`Error·in·GetIsRepoDeletedInDB·in·${path.basename(__filename)}:·${JSON.stringify(exc)}`` with `⏎······"error",⏎······`Error·in·GetIsRepoDeletedInDB·in·${path.basename(⏎········__filename⏎······)}:·${JSON.stringify(exc)}`⏎····`](https://app.codacy.com/gh/thamizhMsys/testRepo6/pullRequest?prid=9650715)
```suggestion
    Logger.log(
```